src/transformers/utils/import_utils.py

```diff
@@ -528,6 +528,26 @@ def get_major_and_minor_from_version(full_version):
     return True
 
 
+@lru_cache
+def is_torch_xpu_available(check_device=False):
+    "Checks if `intel_extension_for_pytorch` is installed and potentially if a XPU is in the environment"
+    if is_ipex_available():
+        import torch
+    else:
+        return False
+
+    import intel_extension_for_pytorch  # noqa: F401
```

Comment:
It's a bit funny to have the condition checking if ipex is available, but then only have `torch` imported in that if/else block

```suggestion
    if not is_ipex_available():
        return False
        
    import torch
    import intel_extension_for_pytorch  # noqa: F401
```

Reply:
Yeah this is something  I was thinking about . Will change